test/Microsoft.AspNetCore.Mvc.IntegrationTests/ArrayModelBinderIntegrationTest.cs

```diff
@@ -327,5 +327,41 @@ private class Person
             Assert.Equal(0, modelState.ErrorCount);
             Assert.True(modelState.IsValid);
         }
+
+        private class PersonWithReadOnly
+        {
+            public string Name { get; set; }
+
+            public string[] Aliases { get; }
```

Comment:
Do we already have a similar test w/ the property non-`null`? Important thing would be to confirm the `ArrayModelBinder` does not overwrite existing elements w/ new values.